app/backend/src/couchers/servicers/communities.py

```diff
@@ -245,3 +245,83 @@ def LeaveCommunity(self, request, context):
             session.query(ClusterSubscription).filter(ClusterSubscription.user_id == context.user_id).delete()
 
             return empty_pb2.Empty()
+    
+    def ListUserCommunities(self, request, context):
+        with session_scope() as session:
+            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
+            next_node_id = int(request.page_token) if request.page_token else 0
+            user_id = context.user_id if request.user_id == 0 else request.user_id
```

Comment:
You could simplify this to `request.user_id or context.user_id`; if `request.user_id` is 0, it'll be falsey in a boolean context, so it'll fall through to using `context.user_id`